File: Features/oracle-primenumber/contracts/src/test/java/net/corda/samples/oracle/states/PrimeStateTests.java
package net.corda.samples.oracle.states;

import net.corda.core.identity.CordaX500Name;
import net.corda.testing.core.TestIdentity;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class PrimeStateTests {

    TestIdentity a = new TestIdentity(new CordaX500Name("Alice", "", "GB"));

    @Test
    public void constructorTest() {
        PrimeState st = new PrimeState(1, 5, a.getParty());

        assertEquals(a.getParty(), st.getRequester());
        assertTrue(st.getParticipants().contains(a.getParty()));
    }
}

File: kotlin-samples/Features/oracle-primenumber/contracts/src/test/kotlin/net/corda/samples/states/PrimeStateTests.kt
package net.corda.samples.states

import net.corda.core.identity.CordaX500Name
import net.corda.testing.core.TestIdentity
import org.junit.Assert
import org.junit.Test

class PrimeStateTests {

    var a = TestIdentity(CordaX500Name("Alice", "", "GB"))

    @Test
    fun constructorTest() {
        val st = PrimeState(1, 5, a.party)
        Assert.assertEquals(a.party, st.requester)
        Assert.assertTrue(st.participants.contains(a.party))
    }
}
